modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -14,10 +14,51 @@ def validate_form_526_submission_values!
         validate_form_526_current_mailing_address_country!
         # ensure homeless information is valid
         validate_form_526_veteran_homelessness!
+        # ensure new address is valid
+        validate_form_526_change_of_address!
         # ensure treament centers information is valid
         validate_form_526_treatments!
       end
 
+      def validate_form_526_change_of_address!
+        return if form_attributes['changeOfAddress'].blank?
+
+        validate_form_526_change_of_address_beginning_date!
+        validate_form_526_change_of_address_ending_date!
+        validate_form_526_change_of_address_country!
+      end
+
+      def validate_form_526_change_of_address_beginning_date!
+        change_of_address = form_attributes['changeOfAddress']
+        date = change_of_address.dig('dates', 'beginningDate')
+        return unless 'TEMPORARY'.casecmp?(change_of_address['typeOfAddressChange'])
+
+        # If the date parse fails, then fall back to the InvalidFieldValue
+        begin
+          return if Date.parse(date) < Time.zone.now
+        rescue
+          raise ::Common::Exceptions::InvalidFieldValue.new('changeOfAddress.dates.beginningDate', date)
+        end
+
+        raise ::Common::Exceptions::InvalidFieldValue.new('changeOfAddress.dates.beginningDate', date)
+      end
+
+      def validate_form_526_change_of_address_ending_date!
```

Comment:
I presume if the endingDate is provided, the beginningDate also will be and that the endingDate should never be before the beginningDate; is that a valid/OK assumption?

Reply:
Sounds reasonable; didn't see anything specific in the Schema Planning spreadsheet about these fields. I did notice that v1 validation on beginningDate included the following--not sure if you think it's wise to add this check as well?
`if Date.parse(change_of_address['beginningDate']) > Time.zone.now`

Reply:
Disregard that, you've already got a check for that.